pageserver/src/layered_repository.rs

```diff
@@ -428,10 +428,10 @@ impl Repository {
 
     pub fn init_attach_timelines(
         &self,
-        timelines: Vec<(ZTimelineId, TimelineMetadata)>,
+        timelines: HashMap<ZTimelineId, TimelineMetadata>,
```

Comment:
Why this change from Vec to HashMap?

Reply:
The main reason is to quickly pass inner values from `TenantTimelineValues`: we now group timelines per tenant to keep the empty tenants visible and this way it's simpler to pass around the values.

I also like it better due to types ensuring better that there are no duplicate timelines in this list.

Reply:
Ok. I didn't see any changes in the callers at a quick glance so I was confused.